assets/js/components/Root/index.js

```diff
@@ -34,7 +34,15 @@ import { FeatureToursDesktop } from '../FeatureToursDesktop';
 import { useFeature } from '../../hooks/useFeature';
 import CurrentSurveyPortal from '../surveys/CurrentSurveyPortal';
 
-export default function Root( { children, registry, viewContext = null } ) {
+export default function Root( props ) {
+	const {
+		children,
+		registry,
+		viewContext = null,
+		withSurveys,
+		withPermissionsModal,
```

Comment:
We don't want to display surveys on non-sitekit pages, right? Plus, and we don't need permissions modals when we are not

Reply:
> We don't want to display surveys on non-sitekit pages, right?

Yes, but simply rendering the `<CurrentSurvey>` doesn't trigger a survey which is the more expensive part since it makes a request. Maybe we should make this controllable by a prop but it seems a bit out of scope here without a concrete need.

> Plus, we don't need permissions modals when we are not on a sitekit page.

I'm not sure about that – the permissions modal will be triggered by any insufficient scopes – not only an on-demand request to create something. Again, maybe we should make this conditional but I think we should move this to a new issue to discuss unless needed here.

> These two props help us to turn off unnecessary functionality and avoid a need to load additional assets.

We should be careful to not be too clever to optimize which assets are loaded based on what we know to be used within the component/module. Instead the module should declare it's dependencies and let WP enqueue them as needed. It sounds like maybe you're trying to avoid loading the `core/user` datastore here but I'm not sure we can do that.

Reply:
:+1: Reverted.